packages/teams-js/src/public/liveShare.ts

```diff
@@ -0,0 +1,128 @@
+import type { AzureConnectionConfig, AzureContainerServices, ITelemetryBaseLogger } from '@fluidframework/azure-client';
+import type { ContainerSchema, IFluidContainer } from '@fluidframework/fluid-static';
+
+import { LiveShareHost } from '../internal/liveShareHost';
+
+/**
+ * Namespace to interact with the Live Share module-specific part of the SDK.
+ *
+ * @beta
+ */
+export namespace liveShare {
+  const LIVE_SHARE_PACKAGE = '@microsoft/live-share';
+  const LIVE_SHARE_HOST = new LiveShareHost();
+  let client: LiveShareClient;
+
+  interface LiveShareClient {
+    new (options?: LiveShareOptions, host?: LiveShareHost);
+    joinContainer(
+      fluidContainerSchema: ContainerSchema,
+      onContainerFirstCreated?: (container: IFluidContainer) => void,
+    ): Promise<{
+      container: IFluidContainer;
+      services: AzureContainerServices;
+      created: boolean;
+    }>;
+  }
+
+  /**
+   * Options used to configure the Live Share client.
+   *
+   * @beta
+   */
+  export interface LiveShareOptions {
+    /**
+     * Optional. Configuration to use when connecting to a custom Azure Fluid Relay instance.
+     */
+    readonly connection?: AzureConnectionConfig;
+
+    /**
+     * Optional. A logger instance to receive diagnostic messages.
+     */
+    readonly logger?: ITelemetryBaseLogger;
+
+    /**
+     * Optional. Function to lookup the ID of the container to use for local testing.
+     *
+     * @remarks
+     * The default implementation attempts to retrieve the containerId from the `window.location.hash`.
+     *
+     * If the function returns 'undefined' a new container will be created.
+     * @returns ID of the container to connect to or `undefined` if a new container should be created.
+     */
+    readonly getLocalTestContainerId?: () => string | undefined;
+
+    /**
+     * Optional. Function to save the ID of a newly created local test container.
+     *
+     * @remarks
+     * The default implementation updates `window.location.hash` with the ID of the newly created
+     * container.
+     * @param containerId The ID of the container that was created.
+     */
+    readonly setLocalTestContainerId?: (containerId: string) => void;
+  }
+
+  /**
+   * Initializes the Live Share client.
+   * @param options Optional. Configuration options passed to the Live Share client.
+   *
+   * @beta
+   */
+  export async function initialize(options?: LiveShareOptions): Promise<void> {
+    if (client) {
```

Comment:
Because you're calling into an async thing between the `if(client)` and when client is actually checked, there's potential for a race condition, even in single-threaded JavaScript context. Is that OK?

Reply:
Good observation. I can protect against that

Reply:
I added logic to protect against multiple calls to initialize()